<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>800</width>
    <height>600</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>800</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>M68K's Virtual Machine</string>
  </property>
  <property name="dockOptions">
   <set>QMainWindow::AnimatedDocks</set>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGraphicsView" name="logo">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>41</width>
      <height>41</height>
     </rect>
    </property>
    <property name="verticalScrollBarPolicy">
     <enum>Qt::ScrollBarAsNeeded</enum>
    </property>
    <property name="interactive">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QLCDNumber" name="Alcd_1">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>10</y>
      <width>61</width>
      <height>31</height>
     </rect>
    </property>
    <property name="lineWidth">
     <number>1</number>
    </property>
    <property name="numDigits">
     <number>4</number>
    </property>
    <property name="mode">
     <enum>QLCDNumber::Hex</enum>
    </property>
    <property name="segmentStyle">
     <enum>QLCDNumber::Filled</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label0">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>10</y>
      <width>51</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Adress Registers:</string>
    </property>
    <property name="wordWrap">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>50</y>
      <width>51</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Data Registers:</string>
    </property>
    <property name="wordWrap">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLCDNumber" name="Alcd_2">
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>10</y>
      <width>61</width>
      <height>31</height>
     </rect>
    </property>
    <property name="numDigits">
     <number>4</number>
    </property>
    <property name="mode">
     <enum>QLCDNumber::Hex</enum>
    </property>
    <property name="segmentStyle">
     <enum>QLCDNumber::Filled</enum>
    </property>
   </widget>
   <widget class="QLCDNumber" name="Alcd_3">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>10</y>
      <width>61</width>
      <height>31</height>
     </rect>
    </property>
    <property name="numDigits">
     <number>4</number>
    </property>
    <property name="mode">
     <enum>QLCDNumber::Hex</enum>
    </property>
    <property name="segmentStyle">
     <enum>QLCDNumber::Filled</enum>
    </property>
   </widget>
   <widget class="QLCDNumber" name="Alcd_4">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>10</y>
      <width>61</width>
      <height>31</height>
     </rect>
    </property>
    <property name="numDigits">
     <number>4</number>
    </property>
    <property name="mode">
     <enum>QLCDNumber::Hex</enum>
    </property>
    <property name="segmentStyle">
     <enum>QLCDNumber::Filled</enum>
    </property>
   </widget>
   <widget class="QLCDNumber" name="Alcd_5">
    <property name="geometry">
     <rect>
      <x>590</x>
      <y>10</y>
      <width>61</width>
      <height>31</height>
     </rect>
    </property>
    <property name="numDigits">
     <number>4</number>
    </property>
    <property name="mode">
     <enum>QLCDNumber::Hex</enum>
    </property>
    <property name="segmentStyle">
     <enum>QLCDNumber::Filled</enum>
    </property>
   </widget>
   <widget class="QLCDNumber" name="Alcd_6">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>10</y>
      <width>61</width>
      <height>31</height>
     </rect>
    </property>
    <property name="numDigits">
     <number>4</number>
    </property>
    <property name="mode">
     <enum>QLCDNumber::Hex</enum>
    </property>
    <property name="segmentStyle">
     <enum>QLCDNumber::Filled</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>10</y>
      <width>31</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>PC:</string>
    </property>
   </widget>
   <widget class="QLCDNumber" name="PClcd">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>10</y>
      <width>81</width>
      <height>41</height>
     </rect>
    </property>
    <property name="lineWidth">
     <number>2</number>
    </property>
    <property name="mode">
     <enum>QLCDNumber::Dec</enum>
    </property>
    <property name="intValue" stdset="0">
     <number>0</number>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="codeEditor">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>90</y>
      <width>271</width>
      <height>331</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="compileButton">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>490</y>
      <width>98</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Build</string>
    </property>
   </widget>
   <widget class="QPushButton" name="runButton">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>460</y>
      <width>98</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Run</string>
    </property>
    <property name="checkable">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="stepButton">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>430</y>
      <width>98</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Step</string>
    </property>
    <property name="checkable">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="resetButton">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>520</y>
      <width>98</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Reset</string>
    </property>
    <property name="checkable">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="plainTextEdit">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>430</y>
      <width>681</width>
      <height>121</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="openButton">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>60</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Open File</string>
    </property>
   </widget>
   <widget class="QPushButton" name="saveButton">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>60</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Save File</string>
    </property>
   </widget>
   <widget class="QLCDNumber" name="CClcd_0">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>90</y>
      <width>21</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>8</pointsize>
     </font>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="lineWidth">
     <number>1</number>
    </property>
    <property name="numDigits">
     <number>1</number>
    </property>
    <property name="mode">
     <enum>QLCDNumber::Hex</enum>
    </property>
    <property name="segmentStyle">
     <enum>QLCDNumber::Filled</enum>
    </property>
    <property name="value" stdset="0">
     <double>0.000000000000000</double>
    </property>
   </widget>
   <widget class="QLCDNumber" name="Alcd_0">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>10</y>
      <width>61</width>
      <height>31</height>
     </rect>
    </property>
    <property name="lineWidth">
     <number>1</number>
    </property>
    <property name="numDigits">
     <number>4</number>
    </property>
    <property name="mode">
     <enum>QLCDNumber::Hex</enum>
    </property>
    <property name="segmentStyle">
     <enum>QLCDNumber::Filled</enum>
    </property>
   </widget>
   <widget class="QLCDNumber" name="Alcd_7">
    <property name="geometry">
     <rect>
      <x>730</x>
      <y>10</y>
      <width>61</width>
      <height>31</height>
     </rect>
    </property>
    <property name="numDigits">
     <number>4</number>
    </property>
    <property name="mode">
     <enum>QLCDNumber::Hex</enum>
    </property>
    <property name="segmentStyle">
     <enum>QLCDNumber::Filled</enum>
    </property>
   </widget>
   <widget class="QLCDNumber" name="Dlcd_6">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>50</y>
      <width>61</width>
      <height>31</height>
     </rect>
    </property>
    <property name="numDigits">
     <number>4</number>
    </property>
    <property name="mode">
     <enum>QLCDNumber::Hex</enum>
    </property>
    <property name="segmentStyle">
     <enum>QLCDNumber::Filled</enum>
    </property>
   </widget>
   <widget class="QLCDNumber" name="Dlcd_1">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>50</y>
      <width>61</width>
      <height>31</height>
     </rect>
    </property>
    <property name="lineWidth">
     <number>1</number>
    </property>
    <property name="numDigits">
     <number>4</number>
    </property>
    <property name="mode">
     <enum>QLCDNumber::Hex</enum>
    </property>
    <property name="segmentStyle">
     <enum>QLCDNumber::Filled</enum>
    </property>
   </widget>
   <widget class="QLCDNumber" name="Dlcd_0">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>50</y>
      <width>61</width>
      <height>31</height>
     </rect>
    </property>
    <property name="lineWidth">
     <number>1</number>
    </property>
    <property name="numDigits">
     <number>4</number>
    </property>
    <property name="mode">
     <enum>QLCDNumber::Hex</enum>
    </property>
    <property name="segmentStyle">
     <enum>QLCDNumber::Filled</enum>
    </property>
   </widget>
   <widget class="QLCDNumber" name="Dlcd_7">
    <property name="geometry">
     <rect>
      <x>730</x>
      <y>50</y>
      <width>61</width>
      <height>31</height>
     </rect>
    </property>
    <property name="numDigits">
     <number>4</number>
    </property>
    <property name="mode">
     <enum>QLCDNumber::Hex</enum>
    </property>
    <property name="segmentStyle">
     <enum>QLCDNumber::Filled</enum>
    </property>
   </widget>
   <widget class="QLCDNumber" name="Dlcd_5">
    <property name="geometry">
     <rect>
      <x>590</x>
      <y>50</y>
      <width>61</width>
      <height>31</height>
     </rect>
    </property>
    <property name="numDigits">
     <number>4</number>
    </property>
    <property name="mode">
     <enum>QLCDNumber::Hex</enum>
    </property>
    <property name="segmentStyle">
     <enum>QLCDNumber::Filled</enum>
    </property>
   </widget>
   <widget class="QLCDNumber" name="Dlcd_3">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>50</y>
      <width>61</width>
      <height>31</height>
     </rect>
    </property>
    <property name="numDigits">
     <number>4</number>
    </property>
    <property name="mode">
     <enum>QLCDNumber::Hex</enum>
    </property>
    <property name="segmentStyle">
     <enum>QLCDNumber::Filled</enum>
    </property>
   </widget>
   <widget class="QLCDNumber" name="Dlcd_4">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>50</y>
      <width>61</width>
      <height>31</height>
     </rect>
    </property>
    <property name="numDigits">
     <number>4</number>
    </property>
    <property name="mode">
     <enum>QLCDNumber::Hex</enum>
    </property>
    <property name="segmentStyle">
     <enum>QLCDNumber::Filled</enum>
    </property>
   </widget>
   <widget class="QLCDNumber" name="Dlcd_2">
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>50</y>
      <width>61</width>
      <height>31</height>
     </rect>
    </property>
    <property name="numDigits">
     <number>4</number>
    </property>
    <property name="mode">
     <enum>QLCDNumber::Hex</enum>
    </property>
    <property name="segmentStyle">
     <enum>QLCDNumber::Filled</enum>
    </property>
   </widget>
   <widget class="QLabel" name="lblCC_0">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>90</y>
      <width>21</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>18</pointsize>
     </font>
    </property>
    <property name="text">
     <string>X</string>
    </property>
   </widget>
   <widget class="QLabel" name="lblCC_1">
    <property name="geometry">
     <rect>
      <x>600</x>
      <y>90</y>
      <width>21</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>18</pointsize>
     </font>
    </property>
    <property name="text">
     <string>N</string>
    </property>
   </widget>
   <widget class="QLCDNumber" name="CClcd_1">
    <property name="geometry">
     <rect>
      <x>620</x>
      <y>90</y>
      <width>21</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>8</pointsize>
     </font>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="lineWidth">
     <number>1</number>
    </property>
    <property name="numDigits">
     <number>1</number>
    </property>
    <property name="mode">
     <enum>QLCDNumber::Hex</enum>
    </property>
    <property name="segmentStyle">
     <enum>QLCDNumber::Filled</enum>
    </property>
    <property name="value" stdset="0">
     <double>0.000000000000000</double>
    </property>
   </widget>
   <widget class="QLCDNumber" name="CClcd_2">
    <property name="geometry">
     <rect>
      <x>670</x>
      <y>90</y>
      <width>21</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>8</pointsize>
     </font>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="lineWidth">
     <number>1</number>
    </property>
    <property name="numDigits">
     <number>1</number>
    </property>
    <property name="mode">
     <enum>QLCDNumber::Hex</enum>
    </property>
    <property name="segmentStyle">
     <enum>QLCDNumber::Filled</enum>
    </property>
    <property name="value" stdset="0">
     <double>0.000000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="lblCC_2">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>90</y>
      <width>21</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>18</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Z</string>
    </property>
   </widget>
   <widget class="QLCDNumber" name="CClcd_3">
    <property name="geometry">
     <rect>
      <x>720</x>
      <y>90</y>
      <width>21</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>8</pointsize>
     </font>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="lineWidth">
     <number>1</number>
    </property>
    <property name="numDigits">
     <number>1</number>
    </property>
    <property name="mode">
     <enum>QLCDNumber::Hex</enum>
    </property>
    <property name="segmentStyle">
     <enum>QLCDNumber::Filled</enum>
    </property>
    <property name="value" stdset="0">
     <double>0.000000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="lblCC_3">
    <property name="geometry">
     <rect>
      <x>700</x>
      <y>90</y>
      <width>21</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>18</pointsize>
     </font>
    </property>
    <property name="text">
     <string>V</string>
    </property>
   </widget>
   <widget class="QLCDNumber" name="CClcd_4">
    <property name="geometry">
     <rect>
      <x>770</x>
      <y>90</y>
      <width>21</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>8</pointsize>
     </font>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="lineWidth">
     <number>1</number>
    </property>
    <property name="numDigits">
     <number>1</number>
    </property>
    <property name="mode">
     <enum>QLCDNumber::Hex</enum>
    </property>
    <property name="segmentStyle">
     <enum>QLCDNumber::Filled</enum>
    </property>
    <property name="value" stdset="0">
     <double>0.000000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="lblCC_4">
    <property name="geometry">
     <rect>
      <x>750</x>
      <y>90</y>
      <width>21</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>18</pointsize>
     </font>
    </property>
    <property name="text">
     <string>C</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="memoryView">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>130</y>
      <width>241</width>
      <height>291</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="programView">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>90</y>
      <width>251</width>
      <height>331</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>compileButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>compile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>50</x>
     <y>527</y>
    </hint>
    <hint type="destinationlabel">
     <x>5</x>
     <y>525</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>codeEditor</sender>
   <signal>textChanged()</signal>
   <receiver>MainWindow</receiver>
   <slot>firstState()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>26</x>
     <y>243</y>
    </hint>
    <hint type="destinationlabel">
     <x>5</x>
     <y>263</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>compile()</slot>
  <slot>firstState()</slot>
 </slots>
</ui>
